DevelopGitHub/GitHubExtension/GitHubExtention.Preferences.WebApi/Azure/AzureBlobStorageMultipartProvider .cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using GitHubExtention.Preferences.WebApi.Queries;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+namespace GitHubExtention.Preferences.WebApi
+{
+    public class AzureBlobStorageMultipartProvider : MultipartFileStreamProvider
+    {
+        private IAzureContainerQuery _container;
+        private string _fileLocation;
+
+        public AzureBlobStorageMultipartProvider(IAzureContainerQuery container)
+            : base(Path.GetTempPath())
+        {
+            _container = container;
+        }
+
+        public string FileLocation
+        {
+            get { return _fileLocation; }
+            set { _fileLocation = value; }
+        }
+
+        public override Task ExecutePostProcessingAsync()
+        {
+            foreach (var fileData in this.FileData)
+            {
+                byte[] content = File.ReadAllBytes(fileData.LocalFileName);
+                if (content.IsValidImageFormat())
+                {
+                    UpdateBlob(
+                        content, 
+                        fileData.Headers.ContentType.MediaType,
+                        Path.GetExtension(fileData.Headers.ContentDisposition.FileName.Trim('"')));
+                }  
+            }
+
+            return base.ExecutePostProcessingAsync();
+        }
+
+        public void UpdateBlob(byte[] content, string contentType, string extension)
```

Comment:
You can return file location from method UpdateBlob. It's covered by case when after uploading file you will get path to it location.